src/Chapter06/Listing06.22.DefiningCalculatedProperties.cs

```diff
@@ -58,6 +58,16 @@ public string Name
             }
             set
             {
+                // #region EXCLUDE
+#if !NET7_0_OR_GREATER
+                value = value?.Trim() ??
+                    throw new ArgumentException("Value cannot be null, empty or Whitespace");
+#else
+                // #endregion EXCLUDE
+                ArgumentException.ThrowIfNullOrEmpty(value.Trim());
```

Comment:
These two are not equivalent. The pre-net7 case assigns the trimmed value into `value` while the net7 case does not.

Reply:
Agreed... I missed this one. (Also indicates unit tests are flawed.)